units/patch_intent.go

```diff
@@ -349,25 +351,88 @@ func (j *patchIntentProcessor) buildCliPatchDoc(ctx context.Context, patchDoc *p
 			patchDoc.Githash, projectRef.Identifier)
 	}
 
+	summaries, err := getPatchSummaries(patchDoc)
+	if err != nil {
+		return errors.Wrap(err, "error getting patch summary")
+	}
+
+	patchDoc.Patches[0].ModuleName = ""
+	patchDoc.Patches[0].PatchSet.Summary = summaries
+
+	return nil
+}
+
+func getPatchSummaries(patchDoc *patch.Patch) ([]patch.Summary, error) {
 	var reader io.ReadCloser
-	reader, err = db.GetGridFile(patch.GridFSPrefix, patchDoc.Patches[0].PatchSet.PatchFileId)
+	reader, err := db.GetGridFile(patch.GridFSPrefix, patchDoc.Patches[0].PatchSet.PatchFileId)
 	if err != nil {
-		return err
+		return nil, errors.Wrap(err, "error getting patch file")
 	}
 	defer reader.Close()
-	bytes, err := ioutil.ReadAll(reader)
-	if err != nil {
-		return err
+	var patchBytes []byte
+	// if commit queue then mbox format, squash body to make summary
+	if patchDoc.Alias == evergreen.CommitQueueAlias {
+		patchBytes, err = handleFormattedPatch(reader)
+		if err != nil {
+			grip.Debug(message.WrapError(err, message.Fields{
+				"ticket": "problem parsing mbox",
+				"patch":  patchDoc.Id.Hex(),
+				"desc":   patchDoc.Description,
+				"alias":  patchDoc.Alias,
+			}))
+			return nil, errors.Wrap(err, "error parsing formatted patch")
+		}
+		// TODO: in the future this should just error
+		// refresh reader to try again in the case of an old commit queue item
+		reader, err = db.GetGridFile(patch.GridFSPrefix, patchDoc.Patches[0].PatchSet.PatchFileId)
+		if err != nil {
+			return nil, errors.Wrap(err, "error getting patch file")
+		}
 	}
+	if len(patchBytes) == 0 {
+		patchBytes, err = ioutil.ReadAll(reader)
+		if err != nil {
+			return nil, errors.Wrap(err, "error parsing patch")
+		}
+	}
+	return thirdparty.GetPatchSummaries(string(patchBytes))
+}
 
-	summaries, err := thirdparty.GetPatchSummaries(string(bytes))
+func handleFormattedPatch(reader io.Reader) ([]byte, error) {
+	var err error
+	var result []byte
+	stream, err := mbox.CreateMboxStream(reader)
 	if err != nil {
-		return err
+		return nil, errors.Wrap(err, "error creating stream")
+	}
+	if stream == nil {
+		return nil, errors.New("mbox stream is nil")
+	}
+	// iterate through patches
+	for err == nil {
+		msg, err := stream.ReadMessage()
+		if err != nil {
+			if err == io.EOF { // no more patches
+				return result, nil
+			}
+			return nil, errors.Wrap(err, "error reading message")
+		}
+		reader := msg.BodyReader()
+		// iterate through patch body
+		for err == nil {
```

Comment:
will this ever be false?

Reply:
hmm maybe not actually, since we break or return in the case of an error. This could just be an unrestricted while loop then.

Reply:
+1, this should just be `for {`